src/index.ts

```diff
@@ -101,6 +101,16 @@ export default class Collection<Resource extends { [key: string]: any }> {
     return col;
   }
 
+  public map(mapFunction: (resource: Resource) => Resource): Collection<Resource> {
```

Comment:
Does this mean the `map` function always has to return the same type that it was passed in?

Reply:
⚡️ d0edfa5